main.go

```diff
@@ -33,28 +35,39 @@ func main() {
 
 func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, os.Kill)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 	svcErrors := make(chan error, 1)
 
+	// Read config
+	cfg, err := config.Get()
+	if err != nil {
+		return fmt.Errorf("unable to retrieve configuration, error: %w", err)
```

Comment:
do we not log here?

Reply:
(I wouldn't but it seems to be the convention in DP to log errors at all levels)

Reply:
Not logging at all levels is something I am trying pretty fervently to move away from, hence providing solutions to deal 
with logData and stackTrace embedded in errors.
So yes, it is a convention but I'm trying to eradicate it :)

Reply:
yes, I agree with Josh point - one thing we might need to investigate is what happens to the stack traces (I think in some places when we log the error, the stack trace only refers to the place where we log)